### PR TITLE
302 refactor product type ITs

### PR DESCRIPTION
**Pull request by heshamMassoud**
#### Description
1. Remove `productTypes/utils/ProductTypeITUtils` and use the already existing one in
`commons/utils/ProductTypeITUtils`.
2. Move `ProductTypeSyncIT` and split into correct classes and packages: `externalsource/producttypes/ProductTypeSyncIT` and `ctpprojectsource/producttypes/ProductTypeSyncIT`
3. Remove beforecallbacks ITs. These will come in a later PR (https://github.com/commercetools/commercetools-sync-java/pull/345) as unit tests.
4. Relax some assertions on error messages from `.equals` to `.contains`

#### Relevant Issues
#302

### CODE REVIEW

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        // test
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(productTypeDrafts)
+            .toCompletableFuture().join();
+
+        // assertion
+        assertThat(errorMessages).isEmpty();
+        assertThat(exceptions).isEmpty();
+        assertThat(productTypeSyncStatistics).hasValues(2, 1, 0, 0);
+    }
```
Comment (ahmetoz) [inline src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java]: It would be good to make it consistent like other tests.

```suggestion
        assertThat(productTypeSyncStatistics
            .getReportMessage())
            .isEqualTo("Summary: 2 product types were processed in total"
                + " (1 created, 0 updated and 0 failed to sync).");
    }
```

File: src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java [hunk 2]
```diff
@@ -170,28 +163,25 @@ public void sync_WithUpdatedProductType_WithNewAttribute_ShouldUpdateProductType
...
 
         final Optional<ProductType> oldProductTypeAfter = getProductTypeByKey(CTP_TARGET_CLIENT, PRODUCT_TYPE_KEY_1);
 
-        assertThat(oldProductTypeAfter).isNotEmpty();
-        assertAttributesAreEqual(oldProductTypeAfter.get().getAttributes(), asList(
-            ATTRIBUTE_DEFINITION_DRAFT_1,
-            ATTRIBUTE_DEFINITION_DRAFT_2,
-            ATTRIBUTE_DEFINITION_DRAFT_3)
-        );
+        assertThat(oldProductTypeAfter).hasValueSatisfying(productType ->
```
Comment (ahmetoz) [inline src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java]: 👀 👍

File: src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java [hunk 3]
```diff
@@ -31,57 +30,52 @@
...
-     * Deletes product types from source and target CTP projects.
-     * Populates source and target CTP projects with test data.
+     * Deletes product types from target CTP project.
+     * Populates target CTP project with test data.
      */
     @Before
     public void setup() {
-        deleteProductTypesFromTargetAndSource();
-        populateSourceProject();
+        deleteProductTypes(CTP_TARGET_CLIENT);
```
Comment (ahmetoz) [inline src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java]: 👍

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java (hunk 1 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java]: 52bbf00

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .errorCallback((error, throwable) -> {
+                errorMessages.add(error);
+                exceptions.add(throwable);
+            })
+            .build();
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
```
Comment (ahalberkamp) [inline src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java]: Maybe you can move this to the setup method, to make the code more readable

File: src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java [hunk 5]
```diff
@@ -244,34 +230,31 @@ public void sync_WithUpdatedProductType_ChangingAttributeOrder_ShouldUpdateProdu
...
             .of(
                 StringAttributeType.of(),
                 "attr_name_1",
-                ofEnglish("attr_label_updated"),
+                LocalizedString.ofEnglish("attr_label_updated"),
                 true
             )
             .attributeConstraint(AttributeConstraint.NONE)
-            .inputTip(ofEnglish("inputTip_updated"))
+            .inputTip(LocalizedString.ofEnglish("inputTip_updated"))
```
Comment (ahalberkamp) [inline src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java]: why did you change this?

File: src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java (hunk 4 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java]: Actually, I'd prefer to leave it like this, since it could be that there is a new coming test which have different options for the sync. I think it's better if every test is isolated and has it's own state.

File: src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java (hunk 5 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeSyncIT.java]: Good catch!: 
931e8c6
